extensions/data-transfer/portability-data-transfer-koofr/src/main/java/org/datatransferproject/transfer/koofr/photos/KoofrPhotosImporter.java

```diff
@@ -0,0 +1,217 @@
+ * limitations under the License.
+ */
+package org.datatransferproject.transfer.koofr.photos;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+import org.apache.commons.imaging.Imaging;
+import org.apache.commons.imaging.common.ImageMetadata;
+import org.apache.commons.imaging.formats.jpeg.JpegImageMetadata;
+import org.apache.commons.imaging.formats.tiff.TiffImageMetadata;
+import org.apache.commons.imaging.formats.tiff.constants.ExifTagConstants;
+import org.apache.commons.io.IOUtils;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.cloud.storage.TemporaryPerJobDataStore;
+import org.datatransferproject.spi.transfer.idempotentexecutor.IdempotentImportExecutor;
+import org.datatransferproject.spi.transfer.provider.ImportResult;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.spi.transfer.types.DestinationMemoryFullException;
+import org.datatransferproject.spi.transfer.types.InvalidTokenException;
+import org.datatransferproject.transfer.ImageStreamProvider;
+import org.datatransferproject.transfer.koofr.KoofrTransmogrificationConfig;
+import org.datatransferproject.transfer.koofr.common.KoofrClient;
+import org.datatransferproject.types.common.models.photos.PhotoAlbum;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+
+/** Imports albums and photos to Koofr. */
+public class KoofrPhotosImporter
+    implements Importer<TokensAndUrlAuthData, PhotosContainerResource> {
+
+  private final KoofrClient koofrClient;
+  private final TemporaryPerJobDataStore jobStore;
+  private final ImageStreamProvider imageStreamProvider;
+  private final Monitor monitor;
+  private final KoofrTransmogrificationConfig transmogrificationConfig =
+      new KoofrTransmogrificationConfig();
+
+  private final SimpleDateFormat exifDateFormat = new SimpleDateFormat("yyyy:MM:dd HH:mm:ss");
+  private final SimpleDateFormat titleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH.mm.ss ");
+
+  public KoofrPhotosImporter(
+      KoofrClient koofrClient, Monitor monitor, TemporaryPerJobDataStore jobStore) {
+    this.koofrClient = koofrClient;
+    this.imageStreamProvider = new ImageStreamProvider();
+    this.monitor = monitor;
+    this.jobStore = jobStore;
+  }
+
+  @Override
+  public ImportResult importItem(
+      UUID jobId,
+      IdempotentImportExecutor idempotentImportExecutor,
+      TokensAndUrlAuthData authData,
+      PhotosContainerResource resource)
+      throws Exception {
+    // Ensure credential is populated
+    koofrClient.getOrCreateCredential(authData);
```

Comment:
One thing to be wary of is that the current interface of the importer is meant to be reused for multiple jobs (we dont do that just yet, but the idea is we would be able to recall importItem() with a different jobid and authData. 

This could end up in a potential race condition and have the authData improperly set for a job. We definitely need to update the interface at some point (this came up in #882 before and we decided to maintain a map of jobId --> client until we update the interface. Is it possible to do that here as well?